### ISSUE TITLE
Regenerate initial labeler config?

### ISSUE THREAD

**Issue by molly** [bluesky-social/ozone#98]
Is it possible to redo the initial setup step where you register the labeler? I realized partway through setup that I was going to need to use a different URL, so I changed the labeler URL and labeler account handle partway through, but forgot to update the labeler config files before doing the registration.

I was able to use the API to update the `services.atproto_labeler.endpoint` and the `verificationMethods.atproto_label` fields on PLC.directory, but something still seems wrong, as labels on accounts are appearing in the Ozone dashboard but not to a Bluesky account viewing the labeled accounts/posts. I'm wondering if it would be better to just regenerate the whole config to ensure it's correct.

Alternatively, is there an easy way to regenerate the whole data block?

Here are the PLC.directory pages, if that's helpful, and the labeler account:
* https://plc.directory/did:plc:pozgajl56zcftf6nzps46z65/data
* https://plc.directory/did:plc:pozgajl56zcftf6nzps46z65/log/audit
* https://bsky.app/profile/cryptolabeler.w3igg.com

**Comment by bnewbold** [comment]
Sounds like a raw labeler and looking forward to subscribing!

If you are blocked on this and just want to move forward, the easiest thing to do might be just nuking the existing labeler account and starting from scratch with a new one? I mention this b/c it looks like the labeler account doesn't have subscribers/followers/posts yet. If you take that path, I would first change the handle on the existing account so something like `whoops123.bsky.social` to ensure there aren't problems using the handle with a new account (we have a small bug with that), but you should then be able to run through everything again following the quickstart.

We will take a poke at the backend and see if there is anything illuminating there.

**Comment by bnewbold** [comment]
One other theory is that the AppView isn't handling updates to the labeler declaration record correctly yet; entirely deleting the record (which will un-register the labeler) and then re-creating it (which will re-register the labeler) might fix it.

**Comment by molly** [comment]
> One other theory is that the AppView isn't handling updates to the labeler declaration record correctly yet; entirely deleting the record (which will un-register the labeler) and then re-creating it (which will re-register the labeler) might fix it.

Deleting the PLC identity record, you mean?

**Comment by bnewbold** [comment]
I meant deleting the `app.bsky.labeler.service` "labeler declaration record" from the labeler account's atproto repo (the data structure that content like posts is stored in). We don't have a great general-purpose UI for deleting arbitrary records from an account for records like this, but the `com.atproto.repo.deleteRecord` API endpoint to the account's PDS will do it.

**Comment by molly** [comment]
Alright, I gave deleting the record a shot, but it didn't seem to help things. I then took your advice on renaming the old labeler account and creating a fresh one, which allowed me to re-register the labeler. All that seemed to work as expected except... I'm still not able to get labels to appear as expected. I've tried applying the label to [this post](https://bsky.app/profile/itarutomy.bsky.social/post/3kqjiksmvqs2e), the profile, and the account, and am still not seeing any change from my BS account that's subscribed to the labeler. Any thoughts?

**Comment by bnewbold** [comment]
TL;DR: my hypothesis is that you're using nginx (instead-of / in-addion-to Caddy), and it needs websockets enabled for the labeler to work.

----

New labeler account and config look great:

- https://bsky.app/profile/cryptolabeler.w3igg.com
- https://web.plc.directory/did/did:plc:smtbhjnwsvcxzvez2nbf7mar

Can connect to the labeler service:

```
$ http get https://ozone.w3igg.com/xrpc/_health
HTTP/1.1 200 OK
Accept-Ranges: bytes
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 19
Content-Type: application/json; charset=utf-8
Date: Mon, 22 Apr 2024 22:56:31 GMT
ETag: W/"13-ajRhzT4o9j+a4hQOugsTz6LCiO4"
Server: nginx/1.18.0 (Ubuntu)
Strict-Transport-Security: max-age=300
Vary: Accept-Encoding
Via: 1.1 varnish
X-Cache: MISS
X-Cache-Hits: 0
X-Powered-By: Express
X-Served-By: cache-bfi-krnt7300031-BFI
X-Timer: S1713826592.792930,VS0,VE72

{
    "version": "0.1.1"
}
```

but can't connect WebSocket:

```
$ websocat wss://ozone.w3igg.com/xrpc/com.atproto.label.subscribeLabels
websocat: WebSocketError: WebSocketError: Received unexpected status code (404 Not Found)
websocat: error running
```

The HTTP headers say:

```
Server: nginx/1.18.0 (Ubuntu)
```

This is a common gotcha for our PDS setup as well, will update HOSTING docs for both projects to call out this issue.

**Comment by molly** [comment]
oh sheesh, yep, that's almost certainly it.

**Comment by molly** [comment]
Fixing the websocket issue seems to have fixed everything! I'll leave it up to you whether or not to close this issue.

**Comment by bnewbold** [comment]
Great! Thanks for sticking with it.

Updated the HACKING doc to avoid this papercut for others (https://github.com/bluesky-social/ozone/pull/99), will close this for now.
